plasma_framework/contracts/src/exits/utils/BondSize.sol

```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.5.0;
+
+import 'openzeppelin-solidity/contracts/math/SafeMath.sol';
+
+library BondSize {
+    using SafeMath for uint256;
+
+    struct Params {
+        uint256 currentGasPrice;
+        uint256 challengeGasCost;
+        uint256 ewmaFactor;
+        uint256 safetyFactor;
+    }
+
+    function bondSize(Params storage _self) internal view returns (uint256) {
+        return _self.challengeGasCost
+            .mul(_self.currentGasPrice)
+            .mul(_self.safetyFactor)
+            .div(100);
+    }
+
+    function updateGasPrice(Params storage _self) internal {
```

Comment:
1. I think the following interface
```
function updateGasPrice(Params storage _self, uint256 _newGasPrice)
```
might be more easy to understand to user compare to a magic `tx.price` inside here, and would make this a more generic interface.


2. Meanwhile, though might only be personal preference, how do you feel about using the following instead of current design that save the updated state directly:
```
bond.updateGasPrice(tx.gasprice).save();
```
vs
```
bond.updateGasPrice(tx.gasprice);
```

We can do this by the following I guess:
```
function updateGasPrice(Params memory _self) internal pure returns (Params memory)
```
This makes the update function taking in `memory` + make the function pure. So if anyone ever needs this bond to do some logic and does not need to save it in storage, it would be more generic useful.